entrypoint.sh

```diff
@@ -25,9 +26,30 @@ if [ -n "$MYSQL_PORT_3306_TCP" ]; then
 		echo >&2 "  Connecting to DB_HOST ($DB_HOST)"
 		echo >&2 '  instead of the linked mysql container.'
 	fi
-fi
+	: ${DB_USER:='root'}
+	if [ "$DB_USER" = 'root' ]; then
+		: ${DB_PASS:=$MYSQL_ENV_MYSQL_ROOT_PASSWORD}
+	fi
+	: ${DB_PORT:='3306'}
+elif [ -n "$POSTGRES_PORT_5432_TCP" ]; then
+	DB_TYPE=pgsql
+	if [ -z "$DB_HOST" ]; then
+		DB_HOST=$POSTGRES_PORT_5432_TCP_ADDR
+		DB_PORT=$POSTGRES_PORT_5432_TCP_PORT
+		: ${DB_NAME:='postgres'}
```

Comment:
The case is a bit different, IMO, with postgres. The postgres image, as configured by default, has you connect to the database named `postgres`. This was intentional, so it wouldn't failover to `drupal`. The mysql image doesn't seem to ship with a default database that you can connect to and start creating tables in. It must either be set via environment or created manually:

> MYSQL_DATABASE
>
> This variable is optional and allows you to specify the name of a database to be created on image startup. If a user/password was supplied (see below) then that user will be granted superuser access (corresponding to GRANT ALL) to this database.

though the postgres image has a similar, albeit undocumented, option.

> POSTGRES_DB
>
> https://github.com/docker-library/postgres/blob/cb70a79c1a8248d7e0db89d79dfbd08dd481be7f/9.3/docker-entrypoint.sh#L43

how would you feel about: 

```bash
if [[ $DB_TYPE == "pgsql" ]]; then
  : ${DB_NAME:=POSTGRES_ENV_POSTGRES_DB}`
elif [[ $DB_TYPE == "mysql" ]]; then
  : ${DB_NAME:=MYSQL_ENV_MYSQL_DATABASE}`
else
  : ${DB_NAME:='drupal'}
fi
```